code-submissions-clustering-plugin/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/cli/AbstractGraphConstructor.kt

```diff
@@ -0,0 +1,81 @@
+package org.jetbrains.research.code.submissions.clustering.cli
+
+import com.intellij.openapi.application.ApplicationStarter
+import com.xenomachina.argparser.ArgParser
+import org.jetbrains.research.code.submissions.clustering.load.context.builder.gumtree.GumTreeGraphContextBuilder
+import org.jetbrains.research.code.submissions.clustering.model.Language
+import org.jetbrains.research.code.submissions.clustering.model.SubmissionsGraph
+import org.jetbrains.research.code.submissions.clustering.util.createFolder
+import org.jetbrains.research.code.submissions.clustering.util.writeToBinary
+import org.jetbrains.research.code.submissions.clustering.util.writeToCsv
+import org.jetbrains.research.code.submissions.clustering.util.writeToString
+import java.nio.file.Paths
+import java.util.logging.Logger
+
+abstract class AbstractGraphConstructor : ApplicationStarter {
+    protected val logger: Logger = Logger.getLogger(javaClass.name)
+    private var toBinary: Boolean = false
+    private var toCSV: Boolean = false
+    private lateinit var language: Language
+    private lateinit var outputPath: String
+
+    protected fun <T : AbstractGraphConstructionArgs> parseArgs(
+        args: MutableList<String>,
+        argsClassConstructor: (ArgParser) -> T,
+        block: T.() -> Unit
```

Comment:
Could you please rename this argument and give more detailed name?

Reply:
Slightly changed the function instead